packages/framework/src/client.ts

```diff
@@ -620,14 +621,20 @@ export class Client {
   }
 
   private async compileControls(templateControls: Record<string, unknown>, event: Event) {
-    const templateString = this.templateEngine.parse(JSON.stringify(templateControls));
+    try {
+      const templateString = this.templateEngine.parse(JSON.stringify(templateControls));
 
-    const compiledString = await this.templateEngine.render(templateString, {
-      ...(event.payload || event.data),
-      subscriber: event.subscriber,
-    });
+      const compiledString = await this.templateEngine.render(templateString, {
+        payload: event.payload || event.data,
+        subscriber: event.subscriber,
+        // Backwards compatibility, for allowing usage of variables without namespace (e.g. `{{name}}` instead of `{{payload.name}}`)
+        ...(event.payload || event.data),
+      });
 
-    return JSON.parse(compiledString);
+      return JSON.parse(compiledString);
+    } catch (error) {
+      throw new CompilingStepControlInvalidError(event.workflowId, event.stepId);
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** Even though we're not consuming it in the UI right now, we should include the `error` on the `data` attribute of the error to ensure that incorrect Liquid syntax will be reported to the Worker engine in the event of incorrect syntax finding it's way into template parsing.‏